lib/kadai-core/src/main/java/io/kadai/common/internal/KadaiEngineImpl.java

```diff
@@ -128,14 +131,34 @@ public class KadaiEngineImpl implements KadaiEngine {
   protected SqlSessionManager sessionManager;
   protected Connection connection;
 
+  private Map<String, String> filterKadaiProperties(Map<String, String> properties) {
+    Map<String, String> filteredProperties = new HashMap<>();
+
+    for (Map.Entry<String, String> entry : properties.entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+
+      if (key.startsWith("kadai.") && !key.toLowerCase().contains("password")) {
+
+        filteredProperties.put(key, value);
+      }
+    }
+
+    return filteredProperties;
+  }
+
   protected KadaiEngineImpl(
       KadaiConfiguration kadaiConfiguration,
       ConnectionManagementMode connectionManagementMode,
       TransactionFactory transactionFactory)
       throws SQLException {
+    Builder builder = new Builder(kadaiConfiguration);
+    builder.setProperties(filterKadaiProperties(kadaiConfiguration.getProperties()));
+    KadaiConfiguration copiedKadaiConfiguration = builder.build();
```

Comment:
Why do you need to instantiate a whole new `KadaiConfiguration`?

Isn't it enough to just modify the `KadaiConfiguration.toString()`method to filter out the properties?

![grafik](https://github.com/user-attachments/assets/078edfa3-e899-4c62-9446-0f29d9e8471c)

Reply:
But the toString method can potentially be used by other methods, that require all properties unfiltered to be given as string

Reply:
But is it now used?

Reply:
I don't think it is used right now yet

Reply:
I think that it is safe to assume that changing the output in the toString() method is not dangerous then, what do you think?